web3/manager.py

```diff
@@ -198,43 +208,66 @@ async def _coro_make_request(
         self.logger.debug(f"Making request. Method: {method}")
         return await request_func(method, params)
 
+    #
+    # formatted_response parses and validates JSON-RPC responses
+    # for expected properties (result or an error) with the expected types.
+    # Responses are inconsistent so allow for missing required JSON-RPC properties
+    #
     @staticmethod
     def formatted_response(
         response: RPCResponse,
         params: Any,
         error_formatters: Optional[Callable[..., Any]] = None,
         null_result_formatters: Optional[Callable[..., Any]] = None,
     ) -> Any:
+        if "jsonrpc" in response and response["jsonrpc"] != "2.0":
+            _raise_bad_response_format(response, '"jsonrpc" must equal "2.0"')
+
+        if "id" in response and not isinstance(response["id"], (str, int, None)):
+            _raise_bad_response_format(
+                response, '"id" must be a string, integer or null'
+            )
+
+        # Format and validate errors
         if "error" in response:
+            error = response.get("error")
+            if error is None or isinstance(error, str):
+                raise ValueError(error)
+
+            # Errors may include a code
+            # https://docs.alchemy.com/reference/error-reference#json-rpc-error-codes
+            code = error.get("code")
+            if not isinstance(code, int):
+                _raise_bad_response_format(response, "error['code'] must be an integer")
+            elif code == METHOD_NOT_FOUND:
+                raise MethodUnavailable(error)
+
+            if not type(error.get("message")) in (str, int, None):
```

Comment:
Oh I think I had that as part of the json schema originally, but I dont recall why I had included it. Since the string is explicitly called out here, I'll update this accordingly.